### ISSUE TITLE
Parameterize common 'reform options' 

### ISSUE THREAD

**Issue by MattHJensen** [OpenSourcePolicyCenter/Tax-Calculator#229]
There are a number typical policy changes that we should parameterize for easy use: 

- [ ] Switch on and off all deductions/credits (we can use 'haircut' parameters)
- [ ] Turn deductions into credits
- [ ] Flip refund-ability of credits 
- [ ] Turn exemption into credit
- [ ] More broadly,  we should allow for a switch to a consumption tax system (x-tax) using the sources method (we may break this into another issue later). See [TPC's VAT paper](http://www.urban.org/uploadedpdf/1001533-Methodology-Distributing-VAT.pdf). We would not need to deal with a change in the price level for an x-tax. This interacts with #224. 
- [ ] Dealing with everything in [Lee-Rubio](http://www.rubio.senate.gov/public/index.cfm/files/serve/?File_id=2d839ff1-f995-427a-86e9-267365609942) would give us practice with many of these issues.

**Comment by martinholmer** [comment]
@MattHJensen raised issue #229 in April 2015 and there has been no discussion since then.
Three of the seven boxes have been checked leaving four of the original goals unchecked.

There have been many  advances in the range of reforms able to be characterized in Tax-Calculator in the two years since this issue was raised.

What, if anything, needs to be done with respect to this issue?

**Comment by MattHJensen** [comment]
I just checked updated this issue by checking Option 53 and Lee-Rubio. The other items are still outstanding, and most of them are blocked by data requirements. But I do think they should remain goals for Tax-Calculator. 

- Switches for turning deductions into credits will require https://github.com/open-source-economics/taxdata/issues/32. 

- Flipping the refundability of credits requires case by case imputations. 

- Microsimulation of an X-Tax requires https://github.com/open-source-economics/taxdata/issues/34 and should also probably come after #1482 as it also relies on an imputation of the split between normal and supernormal returns to savings. 

Each of these projects would need some focussed attention to make headway on.
